Userland/Applications/DisplaySettings/ThemesSettingsWidget.cpp

```diff
@@ -54,11 +51,33 @@ ThemesSettingsWidget::ThemesSettingsWidget(bool& background_settings_changed)
     m_cursor_themes_button->on_click = [&](auto) {
         GUI::Process::spawn_or_show_error(window(), "/bin/MouseSettings", Array { "-t", "cursor-theme" });
     };
+
+    GUI::Application::the()->on_theme_change = [&]() {
+        if (GUI::ConnectionToWindowServer::the().is_system_theme_overridden()) {
+            m_themes_combo->clear_selection();
+            auto theme_override = GUI::ConnectionToWindowServer::the().get_system_theme_override();
```

Comment:
This should be used for the conditional check instead of `is_system_theme_overridden` above now